src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/ServiceBufferFileStreamReader.cs

```diff
@@ -183,17 +186,21 @@ private void AssureBufferLength(int newBufferLength)
         /// <paramref name="convertFunc"/>.
         /// </summary>
         /// <param name="offset">Offset into the file to read from</param>
+        /// <param name="rowId">Internal ID of the row to set on all cells in this row</param>
         /// <param name="convertFunc">Function to use to convert the buffer to the target type</param>
         /// <param name="isNullFunc">
         /// If provided, this function will be used to determine if the value is null
         /// </param>
         /// <param name="toStringFunc">Optional function to use to convert the object to a string.</param>
         /// <typeparam name="T">The expected type of the cell. Used to keep the code honest</typeparam>
         /// <returns>The object, a display value, and the length of the value + its length</returns>
-        private FileStreamReadResult ReadCellHelper<T>(long offset, Func<int, T> convertFunc, Func<int, bool> isNullFunc = null, Func<T, string> toStringFunc = null)
+        private FileStreamReadResult ReadCellHelper<T>(long offset, long rowId,
```

Comment:
Minor: If you're going to put parameters on multiple lines, it's probably a good idea to put all all of them on new lines. E.g.: 
```
private FileStreamReadResult ReadCellHelper<T>(
   long offset, 
   long rowId,
   Func<int, T> convertFunc, 
   Func<int, bool> isNullFunc = null,
   Func<T, string> toStringFunc = null)
{		          

```